secboot/bootstrap_container_sb.go

```diff
@@ -76,6 +83,37 @@ func (bc *bootstrappedContainer) RemoveBootstrapKey() error {
 	return nil
 }
 
+func (bc *bootstrappedContainer) RegisterKeyAsUsed(primaryKey []byte, unlockKey []byte) {
+	// secboot unlocking does not fail when it cannot save keys to the kerying. So
+	// we also want to have a similar behavior and just print warnings in this function.
+	devlinks, err := disksDevlinks(bc.devicePath)
+	if err != nil {
+		logger.Noticef("warning: cannot find symlinks for %s: %v", bc.devicePath, err)
+		return
+	}
+
+	var uuidDevlink string
+	for _, devlink := range devlinks {
+		if strings.HasPrefix(devlink, "/dev/disk/by-uuid/") {
+			uuidDevlink = devlink
+			break
+		}
+	}
+	if uuidDevlink == "" {
+		logger.Noticef("warning: missing by-uuid symlink for %s", bc.devicePath)
+		return
+	}
+
+	logger.Debugf("registering kerying keys for %s (%s)", bc.devicePath, uuidDevlink)
+
+	if _, err := unixAddKey("user", fmt.Sprintf("%s:%s:unlock", defaultKeyringPrefix, uuidDevlink), unlockKey, unix.KEY_SPEC_USER_KEYRING); err != nil {
```

Comment:
This is assumes how secboot formats the key `prefix:devicePath:purpose`. It would be nice if later on secboot provides a helper for unlock/aux keys.

For now, maybe a small comment that this key format is actually coming from secboot to avoid confusion when someone is reading this in the future and doesn't know where this format comes from.

Reply:
I would prefer not to proliferate implementation details for things we should not have had in snapd in the first place. So I will add a comment.